openstack/containerinfra/v1/clustertemplates/results.go

```diff
@@ -0,0 +1,89 @@
+package clustertemplates
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/gophercloud/gophercloud"
+)
+
+type commonResult struct {
+	gophercloud.Result
+}
+
+// CreateResult is the response of a Create operations.
+type CreateResult struct {
+	commonResult
+}
+
+// Extract is a function that accepts a result and extracts a cluster-template resource.
+func (r commonResult) Extract() (*ClusterTemplate, error) {
+	var s *ClusterTemplate
+	err := r.ExtractInto(&s)
+	return s, err
+}
+
+// Represents a template for a Cluster Template
+type ClusterTemplate struct {
+	InsecureRegistry    string             `json:"insecure_registry"`
+	Links               []gophercloud.Link `json:"links"`
+	HTTPProxy           string             `json:"http_proxy"`
+	UpdatedAt           time.Time          `json:"-"`
+	FloatingIPEnabled   bool               `json:"floating_ip_enabled"`
+	FixedSubnet         string             `json:"fixed_subnet"`
+	MasterFlavorID      string             `json:"master_flavor_id"`
+	UUID                string             `json:"uuid"`
+	NoProxy             string             `json:"no_proxy"`
+	HTTPSProxy          string             `json:"https_proxy"`
+	TLSDisabled         bool               `json:"tls_disabled"`
+	KeyPairID           string             `json:"keypair_id"`
+	Public              bool               `json:"public"`
+	Labels              map[string]string  `json:"labels"`
+	DockerVolumeSize    int                `json:"docker_volume_size"`
+	ServerType          string             `json:"server_type"`
+	ExternalNetworkID   string             `json:"external_network_id"`
+	ClusterDistro       string             `json:"cluster_distro"`
+	ImageID             string             `json:"image_id"`
+	VolumeDriver        string             `json:"volume_driver"`
+	RegistryEnabled     bool               `json:"registry_enabled"`
+	DockerStorageDriver string             `json:"docker_storage_driver"`
+	APIServerPort       string             `json:"apiserver_port"`
+	Name                string             `json:"name"`
+	CreatedAt           time.Time          `json:"-"`
+	NetworkDriver       string             `json:"network_driver"`
+	FixedNetwork        string             `json:"fixed_network"`
+	COE                 string             `json:"coe"`
+	FlavorID            string             `json:"flavor_id"`
+	MasterLBEnabled     bool               `json:"master_lb_enabled"`
+	DNSNameServer       string             `json:"dns_nameserver"`
+}
+
+func (r *ClusterTemplate) UnmarshalJSON(b []byte) error {
```

Comment:
For Senlin, you mentioned that it's possible for time values to be returned as both `""` and `null`. That's why we need a special `UnmarshalJSON` method.

Are you sure Magnum does the same thing? If Magnum returns empty times as `null` (which would be the normal behavior), then we can remove this entirely. `time.RFC3339` is the standard time format which Go will correctly unmarshal without any help.

If you are unsure, I recommend removing this.

Reply:
Magnum is using ISO-8601, so shouldn't return both "" and null. Can test some more on it but time.RFC3339 should unmarshal properly